### PR TITLE
Fix-up cairo-threads example now that ImageSurface does not implement…

### LINKED ISSUE #218: Rewrite cairo example
Needed since https://github.com/gtk-rs/cairo/pull/233

cc @sdroege

### PR DESCRIPTION
**Pull request by sdroege**
… Send anymore

Fixes #218.

### CODE REVIEW

**Comment by sdroege** [comment]
Please review first and then I'll later add some more comments about all those changes (and update the old comments).

File: src/bin/cairo_threads.rs [hunk 1]
```diff
@@ -180,15 +238,20 @@ fn draw_image_if_dirty(cr: &Context, image: &ImageSurface, origin: (i32, i32),
...
-        let cr = Context::new(&image_dup);
-        cr.set_source_surface(image, 0.0, 0.0);
-        cr.paint();
+        let surface = ImageSurface::create_for_data(holder, Format::Rgb24, WIDTH, HEIGHT, 4 * WIDTH).expect("Can't create surface");
+        func(&surface);
     }
-
-    image_dup
+    // We have a unique reference to `image` again and return it
+    returner.unwrap()
```
Comment (GuillaumeGomez) [inline src/bin/cairo_threads.rs]: Please use `expect`.

Comment (EPashkin) [reply src/bin/cairo_threads.rs]: expect called in unwrap implementation

**Comment by EPashkin** [comment]
@sdroege LFGM, Thanks
Can you also fix warning?
```
warning: unused import: `gdk::prelude`
  --> src\bin\transparent_main_window.rs:10:5
   |
10 | use gdk::prelude::*;
   |     ^^^^^^^^^^^^
   |
   = note: #[warn(unused_imports)] on by default

```

File: src/bin/cairo_threads.rs [hunk 2]
```diff
@@ -116,7 +166,13 @@ fn build_ui(application: &gtk::Application) {
     area.connect_draw(clone!(cell => move |_, cr| {
         let (ref images, ref origins, _) = *cell.borrow();
         for (image, origin) in images.iter().zip(origins.iter()) {
-            draw_image_if_dirty(&cr, image, *origin, (width, height));
+            let mut image_storage = image.lock().unwrap();
+
+            // Temporarily give ownership of the current image to GTK and then
```
Comment (EPashkin) [inline src/bin/cairo_threads.rs]: maybe "to cairo" ?

Comment (sdroege) [reply src/bin/cairo_threads.rs]: Both :)

**Comment by sdroege** [comment]
Ok, this should be good to go now

**Comment by GuillaumeGomez** [comment]
Thanks!

**Pull request merged**